### PR TITLE
Update cssparser to 0.18

### PR DESCRIPTION
**Pull request by SimonSapin**
~Do not merge yet, depends on https://github.com/servo/rust-cssparser/pull/171.~

---
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/17820)

### CODE REVIEW

File: components/selectors/parser.rs [hunk 1]
```diff
@@ -1231,16 +1232,17 @@ fn parse_qualified_name<'i, 't, P, E, Impl>
         },
         Ok(Token::Delim('*')) => {
             let position = input.position();
-            match input.next_including_whitespace() {
+            // FIXME: remove clone() when lifetimes are non-lexical
+            match input.next_including_whitespace().map(|t| t.clone()) {
```
Comment (emilio) [inline components/selectors/parser.rs]: nit: `cloned()`.

File: components/selectors/parser.rs [hunk 2]
```diff
@@ -1231,16 +1232,17 @@ fn parse_qualified_name<'i, 't, P, E, Impl>
...
                 Ok(Token::Delim('|')) => {
                     explicit_namespace(input, QNamePrefix::ExplicitAnyNamespace)
                 }
                 result => {
                     input.reset(position);
                     if in_attr_selector {
                         match result {
-                            Ok(t) => Err(ParseError::Basic(BasicParseError::UnexpectedToken(t))),
-                             Err(e) => Err(ParseError::Basic(e)),
+                            Ok(t) => Err(ParseError::Basic(BasicParseError::UnexpectedToken(t.clone()))),
```
Comment (emilio) [inline components/selectors/parser.rs]: And, I think that means this clone is not needed?

File: components/selectors/parser.rs [hunk 3]
```diff
@@ -1603,27 +1605,28 @@ fn parse_one_simple_selector<'i, 't, P, E, Impl>(parser: &P,
     where P: Parser<'i, Impl=Impl, Error=E>, Impl: SelectorImpl
 {
     let start_position = input.position();
-    match input.next_including_whitespace() {
+    // FIXME: remove clone() when lifetimes are non-lexical
+    match input.next_including_whitespace().map(|t| t.clone()) {
```
Comment (emilio) [inline components/selectors/parser.rs]: I think you should be able to use `cloned()` here too.

File: components/style/gecko/media_queries.rs [hunk 4]
```diff
@@ -474,47 +474,55 @@ impl Expression {
                          -> Result<Self, ParseError<'i>> {
         input.expect_parenthesis_block()?;
         input.parse_nested_block(|input| {
-            let ident = input.expect_ident()?;
-
-            let mut flags = 0;
-            let result = {
-                let mut feature_name = &*ident;
+            // FIXME: remove extra indented block when lifetimes are non-lexical
```
Comment (emilio) [inline components/style/gecko/media_queries.rs]: Do we have an ETA on this? :(

**Review by emilio — APPROVED** [review]
Looks good... I don't love the fact that the code looks uglier in many places, but... r=me

File: components/selectors/parser.rs (hunk 1 above)
Comment (SimonSapin) [inline components/selectors/parser.rs]: `cloned` exists on `Option` but not on `Result`, unfortunately :(

File: components/style/gecko/media_queries.rs (hunk 4 above)
Comment (SimonSapin) [inline components/style/gecko/media_queries.rs]: There’s a draft RFC and prototype: http://smallcultfollowing.com/babysteps/blog/2017/07/11/non-lexical-lifetimes-draft-rfc-and-prototype-available/ . I’m hopeful it’ll land in stable in 2018.

File: components/selectors/parser.rs (hunk 2 above)
Comment (SimonSapin) [inline components/selectors/parser.rs]: Fixed.

**Comment by SimonSapin** [comment]
https://treeherder.mozilla.org/#/jobs?repo=try&revision=5163e16a7e44196366da5a13c90cf8d91f4d0589

@bors-servo r=emilio

**Pull request merged**